reduction_scripts/reduce_data.py

```diff
@@ -896,99 +936,178 @@ def run_save_3dcube(metadata, prev_suffix, curr_suffix, **args):
+                obs_mode = "ns"
+            elif pywifes.is_subnodshuffle(data_dir + sci_filename):
+                obs_mode = "ns"
             else:
-                pass
+                obs_mode = "class"
+
+            # Read the JSON file
+            proc_steps = load_config_file(f"params_{obs_mode}.json")
+
+            # Create data products directory structure
+            out_dir = os.path.join(working_dir, f"data_products/intermediate/{arm}")
+
+            os.makedirs(out_dir, exist_ok=True)
+
+            calib_prefix = os.path.join(out_dir, f"wifes_{arm}")
+
+            # Some WiFeS specific things
+            my_data_hdu = 0
+
+            # SET SKIP ALREADY DONE FILES ?
+            skip_done = False
+
+            # ------------------------------------------------------------------------
+            # NAMES FOR MASTER CALIBRATION FILES!!!
+            # ------------------------------------------------------------------------
+
+            superbias_fn = "%s_superbias.fits" % calib_prefix
+            superbias_fit_fn = "%s_superbias_fit.fits" % calib_prefix
+            super_dflat_raw = "%s_super_domeflat_raw.fits" % calib_prefix
+            super_dflat_fn = "%s_super_domeflat.fits" % calib_prefix
+            super_dflat_mef = "%s_super_domeflat_mef.fits" % calib_prefix
+            super_tflat_raw = "%s_super_twiflat_raw.fits" % calib_prefix
+            super_tflat_fn = "%s_super_twiflat.fits" % calib_prefix
+            super_tflat_mef = "%s_super_twiflat_mef.fits" % calib_prefix
+            slitlet_def_fn = "%s_slitlet_defs.pkl" % calib_prefix
+            wsol_out_fn = "%s_wave_soln.fits" % calib_prefix
+            wire_out_fn = "%s_wire_soln.fits" % calib_prefix
+            flat_resp_fn = "%s_resp_mef.fits" % calib_prefix
+            calib_fn = "%s_calib.pkl" % calib_prefix
+            tellcorr_fn = "%s_tellcorr.pkl" % calib_prefix
+
+            # ------------------------------------------------------------------------
+            # RUN THE PROCESSING STEPS
+            # ------------------------------------------------------------------------
+            prev_suffix = None
+            for step in proc_steps[arm]:
+                step_name = step["step"]
+                step_run = step["run"]
+                step_suffix = step["suffix"]
+                step_args = step["args"]
+                func_name = "run_" + step_name
+                func = locals()[func_name]
+                if step_run:
+                    func(
+                        obs_metadata,
+                        prev_suffix=prev_suffix,
+                        curr_suffix=step_suffix,
+                        **step_args,
+                    )
+                    if step_suffix != None:
+                        prev_suffix = step_suffix
+                else:
+                    pass
+        except Exception as exc:
+            print("{arm} skipped, as an error occurred during processing: '{exc}'")
```

Comment:
A format string needs an 'f' at the front, so `print(f"{arm} skipped, as an error occurred during processing: '{exc}'.")`, added a fullstop too.

Reply:
oh yes, sorry for the typos.